tests/test.py

```diff
@@ -400,14 +400,16 @@ def test_int_long(self):
 
     def test_fileish_types(self):
         """Test the constructor does the right thing when given different types"""
-        with FitFile(testfile('Settings.FIT')):
+        with FitFile(testfile('Settings.fit')):
             pass
         with FitFile(open(testfile("Settings.fit"), 'rb')):
             pass
-        with FitFile(open(testfile("Settings.fit"), 'rb').read()):
-            pass
-        with FitFile(io.BytesIO(open(testfile("Settings.fit"), 'rb').read())):
-            pass
+        f = open(testfile("Settings.fit"), 'rb')
```

Comment:
Good catch, would prefer still using a context manager for the open/closing instead of an `f.close()`. Something like:
```python
with open(testfile("Settings.fit"), 'rb') as f:
    FitFile(f)
with open(testfile("Settings.fit"), 'rb') as f:
    FitFile(f.read())
with open(testfile("Settings.fit"), 'rb') as f:
    FitFile(io.BytesIO(f.read()))
```